content/blogposts/2022/shift-left-wtf-does-it-mean.md

```diff
@@ -0,0 +1,188 @@
+## Code review
+
+We've talked about "shifting left" many of the previous validation points *from* the review stage to a stage earlier in the dev process. But it's important to realize that code review itself is left of the even more final stages of deployment and production monitoring.
+
+It's important that code reviews are thorough, but often the tools don't scale well to larger changesets. As the old adage goes, "make a 10-line change, you'll get 10 comments, but make a 1,000-line change and you'll get a LGTM rubber stamp."
+
+Having precise code navigation capabilities in code review will help reviewers quickly spin up on the structure and content of the changeset and conduct a review that is both thorough *and* timely.
+
+<figcaption>Use go-to-definition and find references to conduct code reviews that are both thorough and efficient.</figcaption>
+<video className="blog-image" title="Find references in code reviews" alt="Use go-to-definition and find references to conduct code reviews that are both thorough and efficient." loop autoPlay muted playsInline>
+  <source src="https://storage.googleapis.com/sourcegraph-assets/blog/shift-left/10-code-reviews.webm" type="video/webm" data-cookieconsent="ignore"/>
+</video>
+
+Another trick is to break up a large changeset into smaller changesets that don't change the behavior of the system until the very last one (e.g., using feature flags). These smaller changes can be more easily reviewed and validated.
+
+Yet another trick is to request review early, when the code is still rough, but the broad strokes are there. If you're doing TDD for a well-defined problem, this might be pushing up the core unit tests and asking to validate the public interface being tested. If you're more prototyping, this might involve pushing up a very hacky implementation and asking for feedback on just the happy-path client-side user interaction.
+
+## The power of tools
+
+Ultimately, "shift left" is a philosophy that guides a set of specific cultural and process changes in the software development lifecycle. Both the general philosophy and the specific changes can be much assisted with the use of tools that enable the developer to take full charge of shipping a new feature or bug fix.
+
+To learn more about dev tools that assist with shifting left and accelerating the dev cycle, check out the following posts:
+
+- An ex-Googler's guide to dev tools
+- A dev's thoughts on dev productivity
+
+Or you can learn more about how Sourcegraph can serve as a dev tools hub for your organization to drive the cultural and process shifts that shifting left entails.
+
+### About the author
+
+_Beyang Liu is the CTO and co-founder of Sourcegraph. Beyang studied Computer Science at Stanford, where he published research in probabilistic graphical models and computer vision at the Stanford AI Lab. You can chat with Beyang on Twitter [@beyang](https://twitter.com/beyang) or our community [Discord](https://discord.com/invite/vqsBW8m5Y8)._
```

Comment:
> Lgtm! I noticed Marcos added https://github.com/sourcegraph/about/pull/5462 section to all his latest blogposts this week, so we may want to bring that in here too. (Seen at bottom of https://about.sourcegraph.com/blog/developer-productivity-thoughts)

Good call, @st0nebraker. I've added it to the end of the post using Marcos' examples ^

Reply:
Beauteous ❇️ !